enhancements/network/ovn-hypershift-live-migration.md

```diff
@@ -0,0 +1,303 @@
+│ lsp stor-node1     │──┘     └───│ lsp stor-node2      │      
+│ options:           │            │ options:            │    
+│  arp_proxy:        │            │   arp_proxy:        │
+│   0a:58:0a:f3:00:00│            │    0a:58:0a:f3:00:00│
+│   169.254.1.1      │            │    169.254.1.1      │
+└────────────────────┘            └─────────────────────┘
+```
+
+**VMs logical switch ports:**
+
+The logical switch port for new VMs will use ovn-k ipam to reserve an IP address,
+and when live-migrating the VM, the LSP address will be re-used.
+
+CNI must avoid setting the IP address on the migration destination pod.
+
+Also the DHCP options will be configured to deliver the address to the VMs
+
+```text
+┌──────────────────────┐     ┌────────────────────┐
+│logical switch node1  │     │logical switch node2│
+┌──────────────────────┐  │     │   ┌──────────────────────┐      
+│ lsp ns-virt-launcher1│──┘     └───│ lsp ns-virt-launcher2│     
+│ dhcpv4_options: 1234 │            │ dhcpv4_options: 1234 │
+│ address:             │            │ address:             │    
+│  0a:58:0a:f4:00:01   │            │  0a:58:0a:f4:00:01   │
+│  10.244.0.8          │            │  10.244.0.8          │
+└──────────────────────┘            └──────────────────────┘
+
+┌─────────────────────────────────┐
+│ dhcp-options 1234               │
+│   lease_time: 3500              │
+│   router: 169.254.1.1           │
+│   dns_server: [kubedns]         │
+│   server_id: 169.254.1.1        │
+│   server_mac: c0:ff:ee:00:00:01 │
+└─────────────────────────────────┘
+```
+
+**Virt-launcher pod address:**
+The CNI will not set an address at virt-launcher pod netns, that address is
+assigned to the VM with the DHCP options from the LSP, this allows to use 
+kubevirt bridge binding with pod networking and still do live migration.
+
+#### IPAM
+
+The point to point routing feature allows an address to be running at a node different from
+the one "owning" the subnet the address is coming from. This will happen
+after VM live migration.
+
+One scenario for live migration is to shut down the node the VMs were migrated
+from, this means that the IPAM node subnet should go back to the pool but since
+the migrated VMs contains IPs from it those IPs should reserved in case the 
+subnet is assigned to a new node.
+
+On that case before assigning the subnet to the node the VMs ips need to be 
+reserved so they don't get assigned to new pods
+
+Another scenario is ovn-kubernetes pods restarting after live migration, on 
+that case ovnkube-master should discover to what ip pool the VM belongs 
+(using an annotation or searching for subnets) and reserve the address.
+
+#### Detecting hypershift VMs and live migration
+
+This enhancement only applies for hypershift kubevirt workers so the rest of 
+virtual machine should not use it, those VMs will be labeled with
+`kubevirt.io/allow-pod-network-bridge-binding-live-migration` by hypershift
```

Comment:
we need some more steps here on how this will exactly work with OVNK. Who labels the pod, is it the pre-migrated pod? Is a new pod spawned in its place during live migration? Do they have the same name, but different UID?

Reply:
Clarified and added some bullets for those labels and annotations.